lib/galaxy/tools/parameters/basic.py

```diff
@@ -2039,7 +2036,7 @@ def single_to_python(value):
             else:
                 return app.model.context.query( app.model.HistoryDatasetAssociation ).get( int( value ) )
 
-        if isinstance(value, str) and value.find(",") > -1:
+        if (isinstance(value, basestring) and value.find(",") > -1):
```

Comment:
This file is not checked for Python3 compatibility yet, but I would suggest to use `six.string_types` instead of `basestring` anyway (same applies for `lib/galaxy/tools/test.py`).

The external parentheses are not needed.